### PR TITLE
prompt: Add the ability to add new shortcuts to awful.prompt

### PR DESCRIPTION
**Pull request by Elv13**
Example:

    awful.prompt.run({ prompt = "Run: ", hooks = {
        {{         },"Return",function(command)
             local result = awful.util.spawn(command)
             mypromptbox[mouse.screen].widget:set_text(type(result) == "string" and result or "")
             return true
        end},
        {{"Mod1"   },"Return",function(command)
             local result = awful.util.spawn(command,{sticky=true})
             mypromptbox[mouse.screen].widget:set_text(type(result) == "string" and result or "")
             return true
        end},
        {{"Shift"  },"Return",function(command)
             local result = awful.util.spawn(command,{ontop=true,floating=true})
             mypromptbox[mouse.screen].widget:set_text(type(result) == "string" and result or "")
             return true
        end}
        }
    },...)

Another old PR undusting

### CODE REVIEW

File: lib/awful/prompt.lua.in [hunk 1]
```diff
@@ -277,6 +299,23 @@ function prompt.run(args, textbox, exe_callback, completion_callback, history_pa
...
         end
 
+        -- User defined cases
+        if hooks[key] then
+            for k,v in ipairs(hooks[key]) do
+                if #modifiers == #v[1] then
+                    local match = true
+                    for k2,v2 in ipairs(v[1]) do
+                        match = match and mod[v2]
+                        -- There is not enough modifiers to add a if then break end
```
Comment (blueyed) [inline lib/awful/prompt.lua.in]: I don't understand this comment?!

**Comment by Elv13** [comment]
I am not back to Awesome yet, but I at least rebased this one. It seem to still work 2 years later. 

/me also need it since my new computer is too fast to just move the mouse the screen X before the client show up ;P

**Comment by psychon** [comment]
Fine by me, I guess, but the docs should have some more useful examples, I think. Something with functions that actually do something.

**Comment by blueyed** [comment]
> but the docs should have some more useful examples, I think

:+1:

**Comment by blueyed** [comment]
I am removing this from the "next" milestone: documentation has to be made more pretty.

Assigning to @Elv13 - please let us know if this is too much nitpicking and cumbersome.

**Comment by blueyed** [comment]
Followed up in https://github.com/awesomeWM/awesome/issues/483 - I am closing this one.

**Pull request closed without merging**